Rubberduck.Core/Navigation/CodeExplorer/CodeExplorerViewModel.cs

```diff
@@ -237,34 +237,33 @@ private void HandleStateChanged(object sender, ParserStateEventArgs e)
 
             IsBusy = _state.Status != ParserState.Pending && _state.Status <= ParserState.ResolvedDeclarations;
 
-            if (e.State != ParserState.ResolvedDeclarations)
+            if (e.State == ParserState.ResolvedDeclarations)
             {
-                return;
+                Synchronize(_state.DeclarationFinder.AllUserDeclarations);
             }
-
-            Synchronize(_state.DeclarationFinder.AllUserDeclarations.ToList());
         }
 
-        private void Synchronize(List<Declaration> declarations)
+        private void Synchronize(IEnumerable<Declaration> declarations)
```

Comment:
Why change from `List` to `IEnumerable` here when you did the opposite earlier? Considering that it's a private procedure, it should be OK to use concretes?

Reply:
It allows the `IEnumerable` to be materialized on the UI thread instead of in the event handler.

Reply:
Seems that merits a comment so a clown isn't tempted to change it in future?

Reply:
Done.